src/client/app/components/csv/MetersCSVUploadComponent.tsx

```diff
@@ -4,107 +4,192 @@
 
 import * as React from 'react';
 import { FormattedMessage } from 'react-intl';
-import { Button, Form, FormGroup, Input, Label } from 'reactstrap';
-import { MODE } from '../../containers/csv/UploadCSVContainer';
-import { MetersCSVUploadProps } from '../../types/csvUploadForm';
+import { Button, Col, Container, Form, FormGroup, Input, Label, Row } from 'reactstrap';
+import { baseApi } from '../../redux/api/baseApi';
+import { useAppDispatch } from '../../redux/reduxHooks';
+import { MetersCSVUploadPreferencesItem } from '../../types/csvUploadForm';
+import { uploadCSVApi } from '../../utils/api';
+import { MetersCSVUploadDefaults } from '../../utils/csvUploadDefaults';
 import { showErrorNotification, showSuccessNotification } from '../../utils/notifications';
+import translate from '../../utils/translate';
 import FormFileUploaderComponent from '../FormFileUploaderComponent';
-import { AppDispatch } from '../../store';
-import { baseApi } from '../../redux/api/baseApi';
-import { connect } from 'react-redux';
-const mapDispatchToProps = (dispatch: AppDispatch) => {
-	return {
-		resetApiCache: () => dispatch(baseApi.util.invalidateTags(['MeterData']))
-	};
-};
-type ResetProp = { resetApiCache: () => void }
-type MetersCsvUploadPropWithCacheDispatch = MetersCSVUploadProps & ResetProp
-class MetersCSVUploadComponent extends React.Component<MetersCsvUploadPropWithCacheDispatch> {
-	private fileInput: React.RefObject<HTMLInputElement>;
+import TooltipHelpComponent from '../TooltipHelpComponent';
+import TooltipMarkerComponent from '../TooltipMarkerComponent';
 
-	constructor(props: MetersCsvUploadPropWithCacheDispatch) {
-		super(props);
-		this.handleSubmit = this.handleSubmit.bind(this);
-		this.handleSetMeterName = this.handleSetMeterName.bind(this);
-		this.fileInput = React.createRef<HTMLInputElement>();
-	}
+/**
+ * Defines the CSV Meters page
+ * @returns CSV Meters page element
+ */
+export default function MetersCSVUploadComponent() {
+	const dispatch = useAppDispatch();
+	const [meterData, setMeterData] = React.useState<MetersCSVUploadPreferencesItem>(MetersCSVUploadDefaults);
+	const [selectedFile, setSelectedFile] = React.useState<File | null>(null);
+	const [isValidCSV, setIsValidCSV] = React.useState<boolean>(false);
 
-	private async handleSubmit(e: React.MouseEvent<HTMLFormElement>) {
-		try {
-			e.preventDefault();
-			const current = this.fileInput.current as HTMLInputElement;
-			const { files } = current;
-			if (files && (files as FileList).length !== 0) {
-				await this.props.submitCSV(files[0]);
-				// TODO Using an alert is not the best. At some point this should be integrated
-				// with react.
-				showSuccessNotification('<h1>SUCCESS</h1>The meter upload was a success.');
+	// This is needed until a React CSV API is created or integrated with the Redux meter API
```

Comment:
Would the team like to put in an issue about the needed CSV updates for React API? If not, I can do it.